### ISSUE TITLE
`Job.WithGcForce` does not work

### ISSUE THREAD

**Issue by timcassell** [dotnet/BenchmarkDotNet#2568]
While working on #2562, I found that `Job.WithGcForce` doesn't do anything, and the Engine calling `ForceGcCleanups = targetJob.ResolveValue(GcMode.ForceCharacteristic, Resolver);` only ever returns the default value `true`. Likely other characteristics also don't work, but I didn't test them.

cc @AndreyAkinshin

**Comment by filzrev** [comment]
Is this issue is still active?
It seems be resolved by recent async refactor PR.

As far as I've tested with existing tests (`MemoryDiagnoserTests::AllocationQuantumIsNotAnIssueForNetCore21Plus`)
`.WithGcForce(false)` setting is applied to `ForceGcCleanups` property and following conditional code works as expected. 
https://github.com/dotnet/BenchmarkDotNet/blob/1c5cc6676edf66af5d13ac8c284f5c6c04ac1515/src/BenchmarkDotNet/Engines/Engine.cs#L301-L307

**Comment by timcassell** [comment]
Yeah it seems to be fixed. I'm not really sure if it was the async refactor that contributed to the fix or not (I don't see how), or if my original attempts were flawed in some way.
